src/Model/ModelManager.php

```diff
@@ -238,6 +239,14 @@ public function reverseTransform(object $object, array $array = []): void
         }
     }
 
+    public function getRealClass(object $object): string
+    {
+        /** @var class-string<T> $class */
+        $class = \get_class($object);
```

Comment:
That's not true.
What if I call `$modelManager->getRealClass(new \DateTime())` ?

Please add a unit test
```
    public function testGetRealClass(): void
    {
        static::assertSame(\DateTime::class, $this->modelManager->getRealClass(new \DateTime()));
    }
```
it will fail.

If you look at 
```
public function getDocumentManager($class): DocumentManager
    {
        if (\is_object($class)) {
            $class = \get_class($class);
        }
        $dm = $this->registry->getManagerForClass($class);
        if (!$dm instanceof DocumentManager) {
            throw new \RuntimeException(sprintf('No document manager defined for class %s', $class));
        }
        return $dm;
    }
```
it throws exception when no document manager is found.
And I'm pretty sure it can happen when no metadata is found.

I would expect to have
```
public function getRealClass(object $object): string
{
    $class = \get_class($object);

    $dm = $this->registry->getManagerForClass($class);
    if (null === $dm) {
         return $class;
    }

    try {
        return $dm->getClassMetadata($class)->getName();
    } catch (SomethingException $e) {
        return $class;
    }
}
```